edx_solutions_api_integration/imports/views.py

```diff
@@ -193,7 +193,17 @@ def _enroll_user(self, data, errors):
         try:
             cohort = add_cohort(course_key, CourseUserGroup.default_cohort_name, CourseCohort.RANDOM)
         except (IntegrityError, ValueError):
-            cohort = get_cohort_by_name(course_key, CourseUserGroup.default_cohort_name)
+            # Sometimes we can get a situation where although the Cohort exists, read replicas
+            # haven't synced up yet and our read for new Cohorts or related models fails when it hits those replicas.
+            # So we retry a couple times until it works.
+            tried, cohort = 0, None
+            while cohort is None:
+                try:
+                    cohort = get_cohort_by_name(course_key, CourseUserGroup.default_cohort_name)
+                except Exception:
```

Comment:
Shouldn't `CourseUserGroup.DoesNotExist` be sufficient here?

Reply:
That's what I thought about IntegrityError earlier, and then a ValueError started getting thrown :p At this point we just catch everything so a long dev cycle doesnt have to happen given the time crunch.

Reply:
Yup, makes sense :P